### PR TITLE
Lesson 19 Wiki Notes

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: Class-19-(Authentication).md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Class 19: Authentication
+
+## In the prep work this week, the students learned
+
+1. What authentication is.
+1. What sessions and cookies are and how sessions and cookies are used to verify a user's identity.
+1. The importance of encryption.
+1. Why developers use one-way encryption to protect users' passwords.
+1. How to use Spring Security in an application to enable authentication.
```
Comment (CarlyLanglois) [inline Class-19-(Authentication).md]: we actually technically dont use spring security so id say something like, "how to write a basic authentication flow in Spring"

File: Class-19-(Authentication).md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Announcements
+
+1. Students are almost done with the class! One class left!
+1. Assignment #4 is due next class!
```
Comment (CarlyLanglois) [inline Class-19-(Authentication).md]: won't this technically not be true for codergirl or if a class meets only once a week?

File: Class-19-(Authentication).md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Large Group Time (Instructor)
+
+### Class 19 Topics That Require Careful Attention
+1. Review hashing and one-way encryption. Remind students that passwords are not stored, but the encrypted password is stored.
+1. 
+
+## Small Group Time: Class 19 Studio (TA Notes)
+
+1. The studio is focused on adding authentication to ``coding-events``.
```
Comment (CarlyLanglois) [inline Class-19-(Authentication).md]: techjobs* the tutorial in the book goes over the process in coding events.
the studio is meant to be an exercise in applying the same things to techjobs - a lot of the code they created in coding events for the user model, authcontroller, etc can be copy/pasta-ed beause it is not application specific really. But! caution them away from that in the interest of getting more comfortable with what they are coding

File: Class-19-(Authentication).md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+1. Students are almost done with the class! One class left!
+1. Assignment #4 is due next class! 
+1. Check with your course manager for any additional announcements.
+
+## Large Group Time (Instructor)
+
+### Class 19 Topics That Require Careful Attention
+1. Review hashing and one-way encryption. Remind students that passwords are not stored, but the encrypted password is stored.
+1.
```
Comment (CarlyLanglois) [inline Class-19-(Authentication).md]: some ideas:
- discuss popular hashing algorithms and how to implement them
- cover salting as a practice - what is it and why it helps 
- a lot of this section of the book covers a tutorial walkthrough of setting up a basic user authentication work flow in coding-events. there are not videos of this process, nor do we present the finished product. Lots of information is delivered in the tutorial, including some new Spring packages and advanced topics that we only introduce briefly. It is important to emphasize for students that it's ok to treat this topic as an introduction. We are not asking them to be security experts at this point, but this is all valuable information for professional applications and getting exposed to it early on is very useful for their careers
- Emphasize the importance of the process, not necessarily knowing the exact syntax to get it done - syntax is easier to google or research!
- We ask the students to add the same auth flow to techjobs for studio work, so they get a chance to practice this again

**Review by CarlyLanglois — COMMENTED** [review]
added some suggested items for you

File: Class-19-(Authentication).md (hunk 1 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Done!

File: Class-19-(Authentication).md (hunk 2 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Very true! Fixed!

File: Class-19-(Authentication).md (hunk 3 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Done!

File: Class-19-(Authentication).md (hunk 4 above)
Comment (gildedgardenia) [reply Class-19-(Authentication).md]: Thank you!

**Pull request merged**